src/System.Collections/src/System/Collections/Generic/Queue.cs

```diff
@@ -349,6 +348,15 @@ private void SetCapacity(int capacity)
             _version++;
         }
 
+        // Increments the index wrapping it if necessary.
+        private void MoveNext(ref int index)
+        {
+            // It is tempting to use the remainder operator here but it is actually much slower 
+            // than a simple comparison and a rarely taken branch.   
+            int tmp = index + 1;
+            index = (tmp == _array.Length) ? 0 : tmp;
+        }
+
```

Comment:
> Wouldn't this be better?

Have you tried it?  I would expect "no", since every read/write on `index` needs to go through that `ref` (hence the use of the `tmp` here), but if you find otherwise and have data to back it up, PRs are welcome. :smile: